oak_functions_enclave_service/src/lib.rs

```diff
@@ -32,22 +32,27 @@ use oak_functions_service::{
         InitializeRequest, InitializeResponse, InvokeRequest, InvokeResponse, LookupDataChunk,
         OakFunctions, PublicKeyInfo, ReserveRequest, ReserveResponse,
     },
-    Observer,
+    Handler, Observer,
 };
 use prost::Message;
 
-pub struct OakFunctionsService<
+pub struct OakFunctionsService<EKH, EP, H>
+where
     EKH: EncryptionKeyHandle + 'static,
     EP: oak_restricted_kernel_sdk::EvidenceProvider,
-> {
+    H: Handler,
```

Comment:
Personally I would not bother with a generic here, and just use a Box+dyn. Even the existing generics are probably overkill.